folding-schemes/src/frontend/mod.rs

```diff
@@ -14,12 +14,16 @@ pub trait FCircuit<F: PrimeField>: Clone + Debug {
     type Params: Debug;
 
     /// returns a new FCircuit instance
-    fn new(params: Self::Params) -> Self;
+    fn new(params: Self::Params) -> Result<Self, Error>;
```

Comment:
Out of curiosity why turning this into a `Result` now?

Reply:
Because there are cases where an error might be needed to be returned due some internal operation at the `new` call. For example at the circom frontend: https://github.com/privacy-scaling-explorations/sonobe/blob/3aa4483962bacb4a530414e9136915012437af4f/folding-schemes/src/frontend/circom/mod.rs#L34
Before we were not needing that specifically in the circom frontend because the r1cs reading was being done at each folding step, but this was extra work that we could avoid, so now it is done only once at the beginning instead of at each fold, and it can return an error thus needs the `Result`.